spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/spring/tx/TransactionalRepositoryConnectionFactory.java

```diff
@@ -95,7 +95,7 @@ public void closeConnection() {
 		} finally {
 			try {
 				if (con != null && con.isActive()) {
-					logger.warn(
```

Comment:
Can you explain why this should be debug instead of warn? Seems like a fairly critical issue with the users code.

Reply:
On second thought, I agree with you. 

I wanted to reduce the loglevel because I've seen this a lot in my logs without it being of any use, as there were other log messages generated in the same transaction by the problem eventually also leading to this. However, It's just one line, and it is, as you say, pretty serious (the explicit commit or rollback must have failed somehow), so let's leave it in. It might actually be worth digging into the cases in which the message shows up.

Reply:
reverted that change.